<launch>
  <arg name="cx"/>
  <arg name="cy"/>
  <arg name="fx"/>
  <arg name="fy"/>

  <arg name="max_vel"/>
  <arg name="max_acc"/>

  <arg name="point_num"/>
  <arg name="point0_x"/>
  <arg name="point0_y"/>
  <arg name="point0_z"/>
  <arg name="point1_x"/>
  <arg name="point1_y"/>
  <arg name="point1_z"/>
  <arg name="point2_x"/>
  <arg name="point2_y"/>
  <arg name="point2_z"/>
  <arg name="point3_x"/>
  <arg name="point3_y"/>
  <arg name="point3_z"/>
  <arg name="point4_x"/>
  <arg name="point4_y"/>
  <arg name="point4_z"/>
  <arg name="point5_x"/>
  <arg name="point5_y"/>
  <arg name="point5_z"/>
  <arg name="point6_x"/>
  <arg name="point6_y"/>
  <arg name="point6_z"/>
  <arg name="point7_x"/>
  <arg name="point7_y"/>
  <arg name="point7_z"/>

  <arg name="odom_topic"/>

  <node pkg="pass_point_plan" name="pass_point_node" type="pass_point_node" output="screen">
  <!-- trajectory optimization -->
    <remap from="/odom_world" to="$(arg odom_topic)"/>
    <param name="optimization/lambda1" value="20.0" type="double"/>
    <param name="optimization/lambda2" value="5.0" type="double"/>
    <param name="optimization/lambda3" value="0.00001" type="double"/>
    <param name="optimization/lambda4" value="0.01" type="double"/>
    <param name="optimization/lambda7" value="100.0" type="double"/>
    <param name="optimization/dist0" value="0.4" type="double"/>
    <param name="optimization/max_vel" value="$(arg max_vel)" type="double"/>
    <param name="optimization/max_acc" value="$(arg max_acc)" type="double"/>

    <param name="optimization/algorithm1" value="15" type="int"/>
    <param name="optimization/algorithm2" value="11" type="int"/>

    <param name="optimization/max_iteration_num1" value="2" type="int"/>
    <param name="optimization/max_iteration_num2" value="300" type="int"/>
    <param name="optimization/max_iteration_num3" value="200" type="int"/>
    <param name="optimization/max_iteration_num4" value="200" type="int"/>

    <param name="optimization/max_iteration_time1" value="0.0005" type="double"/>
    <param name="optimization/max_iteration_time2" value="0.015" type="double"/>
    <param name="optimization/max_iteration_time3" value="0.01" type="double"/>
    <param name="optimization/max_iteration_time4" value="0.01" type="double"/>

    <param name="optimization/order" value="3" type="int"/>
    <param name="optimization/control_points_distance" value="0.5" type="double"/>

    <param name="optimization/waypoint_num" value="$(arg point_num)" type="int"/>
    <param name="optimization/waypoint0_x" value="$(arg point0_x)" type="double"/>
    <param name="optimization/waypoint0_y" value="$(arg point0_y)" type="double"/>
    <param name="optimization/waypoint0_z" value="$(arg point0_z)" type="double"/>
    <param name="optimization/waypoint1_x" value="$(arg point1_x)" type="double"/>
    <param name="optimization/waypoint1_y" value="$(arg point1_y)" type="double"/>
    <param name="optimization/waypoint1_z" value="$(arg point1_z)" type="double"/>
    <param name="optimization/waypoint2_x" value="$(arg point2_x)" type="double"/>
    <param name="optimization/waypoint2_y" value="$(arg point2_y)" type="double"/>
    <param name="optimization/waypoint2_z" value="$(arg point2_z)" type="double"/>
    <param name="optimization/waypoint3_x" value="$(arg point3_x)" type="double"/>
    <param name="optimization/waypoint3_y" value="$(arg point3_y)" type="double"/>
    <param name="optimization/waypoint3_z" value="$(arg point3_z)" type="double"/>
    <param name="optimization/waypoint4_x" value="$(arg point4_x)" type="double"/>
    <param name="optimization/waypoint4_y" value="$(arg point4_y)" type="double"/>
    <param name="optimization/waypoint4_z" value="$(arg point4_z)" type="double"/>
    <param name="optimization/waypoint5_x" value="$(arg point5_x)" type="double"/>
    <param name="optimization/waypoint5_y" value="$(arg point5_y)" type="double"/>
    <param name="optimization/waypoint5_z" value="$(arg point5_z)" type="double"/>
    <param name="optimization/waypoint6_x" value="$(arg point6_x)" type="double"/>
    <param name="optimization/waypoint6_y" value="$(arg point6_y)" type="double"/>
    <param name="optimization/waypoint6_z" value="$(arg point6_z)" type="double"/>
    <param name="optimization/waypoint7_x" value="$(arg point7_x)" type="double"/>
    <param name="optimization/waypoint7_y" value="$(arg point7_y)" type="double"/>
    <param name="optimization/waypoint7_z" value="$(arg point7_z)" type="double"/>

    <param name="bspline/limit_vel" value="$(arg max_vel)" type="double"/>
    <param name="bspline/limit_acc" value="$(arg max_acc)" type="double"/>
    <param name="bspline/limit_ratio" value="1.1" type="double"/>

    <param name="bspline/limit_vel" value="$(arg max_vel)" type="double"/>
    <param name="bspline/limit_acc" value="$(arg max_acc)" type="double"/>
    <param name="bspline/limit_ratio" value="1.1" type="double"/>
  </node>
</launch>
